src/server/utils/devCdnFactory.js

```diff
@@ -0,0 +1,194 @@
+import path from 'path';
+import fs from 'fs';
+import got from 'got';
+import cors from '@fastify/cors';
+import compress from '@fastify/compress';
+import fastifyStatic from '@fastify/static';
+import Fastify from 'fastify';
+import ip from 'ip';
+import ProxyAgent from 'proxy-agent';
+
+const getLocalModuleMap = ({ pathToModuleMap, oneAppDevCdnAddress }) => {
+  const moduleMap = JSON.parse(fs.readFileSync(pathToModuleMap, 'utf8').toString());
+  Object.keys(moduleMap.modules).forEach((moduleName) => {
+    const module = moduleMap.modules[moduleName];
+    module.browser.url = module.browser.url.replace('[one-app-dev-cdn-url]', oneAppDevCdnAddress);
+    module.legacyBrowser.url = module.legacyBrowser.url.replace('[one-app-dev-cdn-url]', oneAppDevCdnAddress);
+    module.node.url = module.node.url.replace('[one-app-dev-cdn-url]', oneAppDevCdnAddress);
+  });
+  return JSON.stringify(moduleMap, null, 2);
+};
+
+const matchPathToKnownRemoteModuleUrl = (
+  incomingRequestPath,
+  remoteModuleBaseUrls
+) => remoteModuleBaseUrls.find((remoteModuleBaseUrl) => {
+  const remoteModuleUrlOrigin = new URL(remoteModuleBaseUrl).origin;
+  const remoteModulePath = remoteModuleBaseUrl.replace(remoteModuleUrlOrigin, '');
+
+  return incomingRequestPath.startsWith(remoteModulePath);
+});
+
+const consumeRemoteRequest = async (remoteModuleMapUrl, hostAddress, remoteModuleBaseUrls) => {
+  try {
+    const response = await got(remoteModuleMapUrl, {
```

Comment:
Why use `got` and not `node-fetch` (like the rest of our systems do)?

Reply:
Im honestly not entirely sure, It might be because of the use for proxyAgent Im not sure if node-fetch supports that natively?

Reply:
to do it apparently we need to use a package https-proxy-agent as long as proxy agent to make it work

Reply:
https://www.npmjs.com/package/node-fetch#custom-agent